### PR TITLE
scaffolding for track features 

### PR DESCRIPTION
**Pull request by cells2numbers**

### CODE REVIEW

File: R/summarize.R [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    summarized <- tracks(population, ...)
+
+  } else {
+    error <- paste0("undefined operation `", operation, "'")
+
+    futile.logger::flog.error(msg = error)
+
+    stop(error)
+  }
+}
```
Comment (0x00b1) [inline R/summarize.R]: Missing newline.

File: R/track-util.R [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+old <- options(stringsAsFactors = FALSE)
+on.exit(options(old), add = TRUE)
+old <- setwd(tempdir())
+on.exit(setwd(old), add = TRUE)
+
+#' @export
+#' @rdname select_helpers
+
+
+add_dist <- function(tracks, group_track){
```
Comment (0x00b1) [inline R/track-util.R]: This should either be camelCase or shortened (to one word).

File: R/track-util.R [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    dplyr::mutate(Track_dX = Location_Center_X.x - Location_Center_X.y) %>% 
+    dplyr::mutate(Track_dY = Location_Center_Y.x - Location_Center_Y.y) %>% 
+    dplyr::select(-Location_Center_X.x, -Location_Center_Y.x) %>%
+    dplyr::rename(Location_Center_X = Location_Center_X.y) %>%
+    dplyr::rename(Location_Center_Y = Location_Center_Y.y) %>% 
+    dplyr::rename(Metadata_timePoint = Metadata_timePoint2) %>%
+    dplyr::mutate(TrackObjects_Distance_Traveled = sqrt(Track_dX^2 + Track_dY^2)) 
+}
+
+get_track_speed <- function(tracks){
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  tracks %>%
+    dplyr::summarize(Track_Length = n(),
+      Track_Speed = sum(TrackObjects_Distance_Traveled, na.rm = TRUE) / (n() - 1), 
+      Track_Speed_max = max(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Speed_X = sum(Track_dX, na.rm = TRUE) / (n() - 1),
+      Track_Speed_Y = sum(Track_dY, na.rm = TRUE) / (n() - 1)) %>%
+    dplyr::select(-Track_Length)
+}
+
+get_track_fmi <- function(tracks) {
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+   dplyr::summarize(Track_Integrated_Distance_Traveled = sum(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Displacement_X = tail(Location_Center_X, n = 1) - Location_Center_X[1],
+      Track_Displacement_Y = tail(Location_Center_Y, n = 1) - Location_Center_Y[1]
+    ) %>% 
+    dplyr::mutate(Track_xFMI = Track_Displacement_X / Track_Integrated_Distance_Traveled, 
+      Track_yFMI = Track_Displacement_Y / Track_Integrated_Distance_Traveled) %>%
+    dplyr::select(-Track_Integrated_Distance_Traveled, -Track_Displacement_X, -Track_Displacement_Y )
+}
+
+get_track_lifetime  <- function(tracks){
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+  tracks %>%
+   dplyr::summarize( 
+      Track_Length = n(), 
+      Track_Life_Time = length(unique(Metadata_timePoint)),
+      Track_One_Cell = length(unique(Metadata_timePoint)) == length(Metadata_timePoint) )  
+}
+
+get_track_angle <- function(tracks){
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+  tracks %>% 
+   dplyr::summarize(Track_Angle = atan2(tail(Location_Center_Y, n = 1) - Location_Center_Y[1], 
+      tail(Location_Center_X, n = 1) - Location_Center_X[1]))
+}
+
+
+get_track_distance <- function(tracks) {
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+  tracks %>% 
+   dplyr::summarize(Track_Integrated_Distance_Traveled = sum(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Distance_Traveled = sqrt( (tail(Location_Center_Y, n = 1) - Location_Center_Y[1] )^2 + 
+          (tail(Location_Center_X, n = 1) - Location_Center_X[1] )^2 )) 
+}
+
+
+get_track_directionality <- function(tracks){
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+  tracks %>% 
+    get_track_distance %>%
+    dplyr::mutate( Track_Directionality = Track_Distance_Traveled / Track_Integrated_Distance_Traveled) %>%
+    dplyr::select(-Track_Distance_Traveled, -Track_Integrated_Distance_Traveled)
+}
+
+
+
+get_track_msd <- function(tracks,t) {
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+ tracks %>% 
+  dplyr::summarize(Track_MSD = (Location_Center_X[t] - Location_Center_X[1])^2 +
+                                (Location_Center_Y[t] - Location_Center_Y[1])^2)
+}
+
+
+get_track_dp <- function(tracks) {
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+  tracks %>%
+    get_track_directionality %>%
+    dplyr::mutate(Track_DP = ceiling(3 * Track_Directionality)) %>%
+    dplyr::select(-Track_Directionality)
+}
+
+
+get_track_chemotactic_index <- function(tracks){
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+
+# returned sector indices
+#
+#   \  3  /
+# 1   \ /   2
+#     / \
+#   /  4  \
+get_track_sectors <- function(tracks){
```
Comment (0x00b1) [inline R/track-util.R]: Omit `get` from identifier. A function that returns a value (and doesn’t modify state) is implicitly a getter.

File: R/track-util.R [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+# make sure to not disturb the environment ------------------------------------
+old <- options(stringsAsFactors = FALSE)
+on.exit(options(old), add = TRUE)
+old <- setwd(tempdir())
+on.exit(setwd(old), add = TRUE)
+
+#' @export
+#' @rdname select_helpers
+
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+    tracks, by = (.dots = c(group_track, "Metadata_timePoint2" = "Metadata_timePoint" ))) %>%
+    dplyr::mutate(Track_dX = Location_Center_X.x - Location_Center_X.y) %>% 
+    dplyr::mutate(Track_dY = Location_Center_Y.x - Location_Center_Y.y) %>% 
+    dplyr::select(-Location_Center_X.x, -Location_Center_Y.x) %>%
+    dplyr::rename(Location_Center_X = Location_Center_X.y) %>%
+    dplyr::rename(Location_Center_Y = Location_Center_Y.y) %>% 
+    dplyr::rename(Metadata_timePoint = Metadata_timePoint2) %>%
+    dplyr::mutate(TrackObjects_Distance_Traveled = sqrt(Track_dX^2 + Track_dY^2)) 
+}
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+get_track_speed <- function(tracks){
+  tracks %>%
+    dplyr::summarize(Track_Length = n(),
+      Track_Speed = sum(TrackObjects_Distance_Traveled, na.rm = TRUE) / (n() - 1), 
+      Track_Speed_max = max(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Speed_X = sum(Track_dX, na.rm = TRUE) / (n() - 1),
+      Track_Speed_Y = sum(Track_dY, na.rm = TRUE) / (n() - 1)) %>%
+    dplyr::select(-Track_Length)
+}
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 16]
```diff
@@ -0,0 +1,142 @@
...
+  s <- tracks %>% 
+   dplyr::summarize(Track_Integrated_Distance_Traveled = sum(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Displacement_X = tail(Location_Center_X, n = 1) - Location_Center_X[1],
+      Track_Displacement_Y = tail(Location_Center_Y, n = 1) - Location_Center_Y[1]
+    ) %>% 
+    dplyr::mutate(Track_xFMI = Track_Displacement_X / Track_Integrated_Distance_Traveled, 
+      Track_yFMI = Track_Displacement_Y / Track_Integrated_Distance_Traveled) %>%
+    dplyr::select(-Track_Integrated_Distance_Traveled, -Track_Displacement_X, -Track_Displacement_Y )
+}
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+get_track_lifetime  <- function(tracks){
+  tracks %>%
+   dplyr::summarize( 
+      Track_Length = n(), 
+      Track_Life_Time = length(unique(Metadata_timePoint)),
+      Track_One_Cell = length(unique(Metadata_timePoint)) == length(Metadata_timePoint) )  
+}
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+get_track_angle <- function(tracks){
+  tracks %>% 
+   dplyr::summarize(Track_Angle = atan2(tail(Location_Center_Y, n = 1) - Location_Center_Y[1], 
+      tail(Location_Center_X, n = 1) - Location_Center_X[1]))
+}
+
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+get_track_distance <- function(tracks) {
+  tracks %>% 
+   dplyr::summarize(Track_Integrated_Distance_Traveled = sum(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Distance_Traveled = sqrt( (tail(Location_Center_Y, n = 1) - Location_Center_Y[1] )^2 + 
+          (tail(Location_Center_X, n = 1) - Location_Center_X[1] )^2 )) 
+}
+
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+get_track_directionality <- function(tracks){
+  tracks %>% 
+    get_track_distance %>%
+    dplyr::mutate( Track_Directionality = Track_Distance_Traveled / Track_Integrated_Distance_Traveled) %>%
+    dplyr::select(-Track_Distance_Traveled, -Track_Integrated_Distance_Traveled)
+}
+
+
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 21]
```diff
@@ -0,0 +1,142 @@
...
+get_track_msd <- function(tracks,t) {
+ tracks %>% 
+  dplyr::summarize(Track_MSD = (Location_Center_X[t] - Location_Center_X[1])^2 +
+                                (Location_Center_Y[t] - Location_Center_Y[1])^2)
+}
+
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 22]
```diff
@@ -0,0 +1,142 @@
...
+get_track_dp <- function(tracks) {
+  tracks %>%
+    get_track_directionality %>%
+    dplyr::mutate(Track_DP = ceiling(3 * Track_Directionality)) %>%
+    dplyr::select(-Track_Directionality)
+}
+
+
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+    dplyr::select(-Track_Angle)
+}
+
+
+# returned sector indices
+#
+#   \  3  /
+# 1   \ /   2
+#     / \
+#   /  4  \
```
Comment (0x00b1) [inline R/track-util.R]: Missing #' @export.

File: R/track-util.R [hunk 24]
```diff
@@ -0,0 +1,142 @@
+#' track helpers
```
Comment (0x00b1) [inline R/track-util.R:1]: I dislike “utils” modules. They are usually a clue that the architecture is amiss.

Comment (0x00b1) [reply R/track-util.R:1]: In this situation, I think it’s indicative of too much decoupling. 

Each function is called exactly once by the same function in the same place (`cytominer::track`). I’d recommend moving these functions into track.R.

File: R/track-util.R (hunk 3 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 4 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 5 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 6 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 7 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 8 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 9 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 10 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 11 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 12 above)
Comment (0x00b1) [reply R/track-util.R]: You expose a track namespace. You don’t need to prefix your identifiers with “track.”

File: R/track-util.R (hunk 24 above)
Comment (0x00b1) [reply R/track-util.R:1]: An additional benefit from moving the functions into track.R: you resolve the `# 'export` issues.

File: tests/testthat/test-track.R [hunk 25]
```diff
@@ -0,0 +1,143 @@
+context("track")
+
+test_that("`track` collapse single cell data to track objects", {
+
```
Comment (0x00b1) [inline tests/testthat/test-track.R]: Omit newline.

File: R/track-util.R (hunk 2 above)
Comment (0x00b1) [reply R/track-util.R]: What about `displace`?

Comment (0x00b1) [inline R/track-util.R]: Missing space between arguments and opening bracket (`{`).

File: R/summarize.R [hunk 26]
```diff
@@ -0,0 +1,24 @@
+#' Summarize data
```
Comment (0x00b1) [inline R/summarize.R]: Where’d this come from? Should this be moved into a separate pull request?

File: R/track.R [hunk 27]
```diff
@@ -0,0 +1,30 @@
...
+#'
+#' @param population ...
+#'
+#' @return track statistics
+#'
+#' @importFrom magrittr %>%
+#' @importFrom magrittr %<>%
+#' @export
+#'
+track <- function(population, group_track, ...) {
```
Comment (0x00b1) [inline R/track.R]: What’s `group_track`?

File: R/track.R [hunk 28]
```diff
@@ -0,0 +1,30 @@
...
+  # process `population`, which is the data you get from CellProfiler
+  tracks <- add_dist(population, group_track)
+  
+  track_feature_list <- list(
```
Comment (0x00b1) [inline R/track.R]: You’re protected by the function namespace you can use a clearer identifier like “features.”

File: R/track.R [hunk 29]
```diff
@@ -0,0 +1,30 @@
...
+    get_track_directionality(tracks),
+    get_track_distance(tracks),
+    get_track_dp(tracks),
+    get_track_fmi(tracks),
+    get_track_lifetime(tracks),
+    get_track_msd(tracks,t = 5),
+    get_track_sectors(tracks),
+    get_track_speed(tracks)) 
+  
+  tracks <-  Reduce(function(...) merge(..., all = TRUE, by = group_track), track_feature_list)
```
Comment (0x00b1) [inline R/track.R]: You should save the allocation by immediately returning.

File: R/track-util.R (hunk 11 above)
Comment (0x00b1) [inline R/track-util.R]: `chemotaxis` might be clearer.

File: R/track-util.R (hunk 4 above)
Comment (0x00b1) [reply R/track-util.R]: Avoid abbreviations unless they are extremely common.

File: R/track-util.R (hunk 9 above)
Comment (0x00b1) [reply R/track-util.R]: Avoid abbreviations unless they are extremely common.

**Review by 0x00b1 — CHANGES_REQUESTED** [review]

**Comment by 0x00b1** [comment]
@shntnu What do you think about this pull request? It requires CellProfiler naming conventions and creates its own feature-specific namespace. Should Cytominer depend on CellProfiler naming like this? Should we introduce feature-specific namespace for other features like geometry or texture?

**Comment by 0x00b1** [comment]
@shntnu If you’re curious about my opinion, I think we shouldn’t depend on CellProfiler naming and I think certain feature areas could necessitate a unique namespace (and motion-estimation might be one).

**Comment by 0x00b1** [comment]
@cells2numbers this is failing on Travis because of the aforementioned export issues.

File: R/summarize.R (hunk 26 above)
Comment (cells2numbers) [reply R/summarize.R]: not needed, will be removed

File: R/track-util.R (hunk 2 above)
Comment (cells2numbers) [reply R/track-util.R]: displace sounds good

File: R/track-util.R (hunk 4 above)
Comment (cells2numbers) [reply R/track-util.R]: you prefer to rename get_track_fmi to forwardMigrationIndex?

File: R/track-util.R (hunk 3 above)
Comment (cells2numbers) [reply R/track-util.R]: all get_track_ functions have been renamed, looks much cleaner now

File: R/track-util.R (hunk 24 above)
Comment (cells2numbers) [reply R/track-util.R:1]: functions are moved into track.R. You recommend not to use a track-utils.R at all?

File: R/track-util.R (hunk 5 above)
Comment (cells2numbers) [reply R/track-util.R]: all renamed

File: R/track-util.R (hunk 9 above)
Comment (cells2numbers) [reply R/track-util.R]: renamed to meanSquaredDisplacement

File: R/track-util.R (hunk 11 above)
Comment (cells2numbers) [reply R/track-util.R]: not my name :) chemotaxis index (CI) is a parameter found in literature

File: R/track.R (hunk 28 above)
Comment (cells2numbers) [reply R/track.R]: done, thx

**Comment by shntnu** [comment]
@0x00b1 
I agree we should avoid tying ourselves to CP names. 

e.g.:

```
get_track_angle <- function(tracks){
  tracks %>% 
   dplyr::summarize(Track_Angle = atan2(tail(Location_Center_Y, n = 1) - Location_Center_Y[1], 
      tail(Location_Center_X, n = 1) - Location_Center_X[1]))
}
```

We can rewrite that using `quo` (starting dplyr 0.6 https://blog.rstudio.org/2017/04/13/dplyr-0-6-0-coming-soon/)

```
get_track_angle <- function(tracks, x = 'Location_Center_X', y = 'Location_Center_Y') {
  tracks %>% 
   dplyr::summarize(track_angle = atan2(tail(quo(x), n = 1) - quo(y)[1], 
      tail(quo(x), n = 1) - quo(x)[1]))
}
```

So the default names will be CP, but it needn't be so.

**Comment by shntnu** [comment]
@0x00b1 I'm less clear about how you would define a unique namespace. Can you provide an example?

File: R/track-util.R (hunk 24 above)
Comment (shntnu) [reply R/track-util.R:1]: @0x00b1 I guided @cells2numbers in that direction, i.e., to use `*-utils.R` but did so because I saw it elsewhere (dplyr). I see the benefit in keeping it in one place (track.R) but wondering if that would make it too cluttered?

File: R/track-util.R (hunk 2 above)
Comment (shntnu) [reply R/track-util.R]: phew!
(dontLikeCamelCase)

File: R/track-util.R (hunk 24 above)
Comment (cells2numbers) [reply R/track-util.R:1]: I removed the track-utils and put all functions in track.R. I suggest to restructure the code with rising  complexity .

File: R/track-util.R (hunk 2 above)
Comment (cells2numbers) [reply R/track-util.R]: hmm, time for a style guide?

**Comment by cells2numbers** [comment]
I moved all functions to track.R and each function is exported with #' @export. Still, travis fails to run the tests. Two questions: 

1. how can I reproduce travis results on my computer? I could install from github and build it, but is there another (shorter) way? 
2. How can I fix the export? 
@0x00b1  @shntnu

File: R/track.R [hunk 30]
```diff
@@ -0,0 +1,403 @@
...
+#'  data <- tibble::data_frame(
+#'    Metadata_timePoint = c(1:5),
+#'    Location_Center_X = c(1, 2, 3, 4, 5),
+#'    Location_Center_Y = c(1, 1, 1, 1, 1),
+#'    TrackObjects_Label = c(rep(1, 5))
+#'  )
+#'  data <- dplyr::group_by_(data,'TrackObjects_Label')
+#'  tracks <- track(data,'TrackObjects_Label')
+#' @export
+track <- function(population, grouping_variable) {
```
Comment (mcquin) [inline R/track.R]: IIRC, we use `strata` to identify grouping variables.

File: R/track.R [hunk 31]
```diff
@@ -0,0 +1,403 @@
+#' Compute track statistics
+#'
+#' @param population, single cell data
+#' @param grouping_variable, column name storing the track label
+#' @return track
+#' @importFrom magrittr %>%
+#' @examples
```
Comment (mcquin) [inline R/track.R]: Nice! 😎

File: R/track.R [hunk 32]
```diff
@@ -0,0 +1,403 @@
...
+#'  data <- dplyr::group_by_(data,'TrackObjects_Label')
+#'  tracks <- track(data,'TrackObjects_Label')
+#' @export
+track <- function(population, grouping_variable) {
+  # process `population`, which is the data you get from CellProfiler
+  tracks <- displace(population, grouping_variable)
+  
+  features <- list(
+    angle(tracks),
+    chemotacticIndex(tracks),
```
Comment (mcquin) [inline R/track.R]: `chromatic_index`

Comment (mcquin) [reply R/track.R]: Use `snake_case` rather than `CamelCase` for variables.

File: R/track.R [hunk 33]
```diff
@@ -0,0 +1,403 @@
...
+    directionality(tracks),
+    distance(tracks),
+    directionalPersistence(tracks),
+    forwardMigrationIndex(tracks),
+    lifeTime(tracks),
+    meanSquaredDisplacement(tracks,tau = 2),
```
Comment (mcquin) [inline R/track.R]: Missing space between arguments.

File: R/track.R [hunk 34]
```diff
@@ -0,0 +1,403 @@
...
+#'    TrackObjects_Label = c(rep(1, 5))
+#'  )
+#'  tracks <- cytominer::displace(data,'TrackObjects_Label')
+#' @importFrom magrittr %>%
+#' @export
+displace <- function(population, group_variable) {
+  dplyr::right_join(
+    population %>%
+      dplyr::select_(.dots = c(group_variable, 'Location_Center_X', 'Location_Center_Y','Metadata_timePoint')) %>%
+      dplyr::mutate(Metadata_timePoint2 =  (Metadata_timePoint - 1) ) %>%
```
Comment (mcquin) [inline R/track.R]: @shntnu We need to make a decision about whether we should or should not use CellProfiler feature names. We should probably force the user to pass the appropriate arguments rather than collect them from some parent data structure.

File: R/track.R [hunk 35]
```diff
@@ -0,0 +1,403 @@
...
+    dplyr::summarize(Track_Length = n(),
+      Track_Speed = sum(TrackObjects_Distance_Traveled, na.rm = TRUE) / (n() - 1), 
+      Track_Speed_max = max(TrackObjects_Distance_Traveled, na.rm = TRUE),
+      Track_Speed_X = sum(Track_dX, na.rm = TRUE) / (n() - 1),
+      Track_Speed_Y = sum(Track_dY, na.rm = TRUE) / (n() - 1)) %>%
+    dplyr::select(-Track_Length)
+  
+}
+
+#' Computer the forward migration index of a track object
```
Comment (mcquin) [inline R/track.R]: Compute

Comment (mcquin) [inline R/track.R]: *of a tracked object

File: R/track.R [hunk 36]
```diff
@@ -0,0 +1,403 @@
...
+#'    Location_Center_X = c(1, 2, 3, 4, 5),
+#'    Location_Center_Y = c(1, 1, 1, 1, 1),
+#'    TrackObjects_Label = c(rep(1, 5))
+#'  )
+#'  tracks <- cytominer::displace(data,'TrackObjects_Label')
+#'  lifeTime <-  cytominer::lifeTime(tracks)
+#'    
+#' @importFrom magrittr %>%
+#' @export
+lifeTime  <- function(tracks) {
```
Comment (mcquin) [inline R/track.R]: `lifetime` like @0x00b1's favorite television channel

File: R/track.R (hunk 34 above)
Comment (shntnu) [reply R/track.R]: I agree that we should not use CellProfiler feature names - see alternative here https://github.com/cytomining/cytominer/pull/45#issuecomment-299316036

File: R/track.R (hunk 32 above)
Comment (cells2numbers) [reply R/track.R]: I started with the snake and was pointed to the camel. Time for a style guide 😄

File: R/track.R (hunk 34 above)
Comment (cells2numbers) [reply R/track.R]: Shantanu mentioned to use quo(x) that is available in the next dplyr release. Q: how can I link is answer?

File: R/track.R (hunk 35 above)
Comment (cells2numbers) [reply R/track.R]: Not sure. The implemented verb is track and the tracks are stored in an on track object. Does that make sense?

File: R/track.R (hunk 32 above)
Comment (shntnu) [reply R/track.R]: I recommend we use http://r-pkgs.had.co.nz/r.html as a starting point for the style guide

File: R/track.R (hunk 34 above)
Comment (shntnu) [reply R/track.R]: @cells2numbers to link to a comment, right click on the "time since comment" next to the username (e.g. "12 days ago")  and copy

**Comment by cells2numbers** [comment]
The code does not fit into the cytominer package as its main purpose is to calculate new feature.  It is moved to its own small package https://github.com/cells2numbers/neutrominer. 

The PR is closed without merging into master
